src/DependencyInjection/Configuration.php

```diff
@@ -19,8 +19,13 @@ class Configuration implements ConfigurationInterface
      */
     public function getConfigTreeBuilder()
     {
-        $treeBuilder = new TreeBuilder();
-        $rootNode = $treeBuilder->root('knp_menu');
+        if (\method_exists(TreeBuilder::class, '__construct')) {
```

Comment:
The detection should be based on `getRootNode` instead.

And I would always call `$treeBuilder = new TreeBuilder('knp_menu');`, as done in many other bundles. Passing the extra argument on older Symfony versions is not an issue.

Reply:
OK, checking for getRootNode() now.

I saw the constructor call in other bundles, but that looks quite smelly - let me know if you insist on this, otherwise I'd keep the cleaner distinction.